spyder/plugins/variableexplorer/widgets/namespacebrowser.py

```diff
@@ -39,10 +40,16 @@
 from spyder.plugins.variableexplorer.widgets.collectionseditor import (
     RemoteCollectionsEditorTableView)
 from spyder.plugins.variableexplorer.widgets.importwizard import ImportWizard
+from spyder.widgets.helperwidgets import FinderLineEdit
 
 
 SUPPORTED_TYPES = get_supported_types()
 
+if PY2:
+    VALID_VARIABLE_CHARS = r"^[^\W]\w*\Z"
+else:
+    VALID_VARIABLE_CHARS = r"^[^\W]\w*\Z"

```

Comment:
Why do these regexp's are exactly the same for Python 2 and 3? Shouldn't they be different?

Reply:
Yep, I just pushed the commit to check the testing. I haven't checked the regex for Python 3

Reply:
Ok, thanks for the clarification @dalthviz!